### PR TITLE
[KOGITO-3094] - Provide data-index-infinispan/mongodb as independent …

### PR DESCRIPTION
**Pull request by spolti**
…images

Signed-off-by: spolti <fspolti@redhat.com>

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by spolti** [comment]
@radtriste for bdd tests, probably needs to be added after operator is able to handle the mongodb infra as well, right?

File: examples/data-index-mongodb.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    apiVersion: kafka.strimzi.io/v1beta1
+    kind: Kafka
+---
+# requires a existing mongodb instance running on the target namespace
+apiVersion: app.kiegroup.org/v1alpha1
+kind: KogitoDataIndex
+metadata:
+  name: data-index
+spec:
+  image: quay.io/spolti/kogito-data-index-mongodb:1.0
```
Comment (sutaakar) [inline examples/data-index-mongodb.yaml:17]:
```suggestion
  image: quay.io/kiegroup/kogito-data-index-mongodb:1.0
```

**Comment by sutaakar** [comment]
> @radtriste for bdd tests, probably needs to be added after operator is able to handle the mongodb infra as well, right?

@spolti Right

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,15 +1,10 @@
 ## Enhancements
 
-- [KOGITO-3270](https://issues.redhat.com/browse/KOGITO-3270) - Operator should support ConfigMap injecting in KogitoServices
-- [KOGITO-2905](https://issues.redhat.com/browse/KOGITO-2905) - Change in image definition in the CRDs
-- [KOGITO-3039](https://issues.redhat.com/browse/KOGITO-3039) - Agnostic infrastructure layer configuration on Operator
-- [KOGITO-2113](https://issues.redhat.com/browse/KOGITO-2113) - Simplify binary build deployment
+- [KOGITO-3094](Provide data-index-infinispan/mongodb as independent images) - Provide data-index-infinispan/mongodb as independent images
+     - To switch to Mongodb persistence provider just provide the `quay.io/kiegroup/kogito-data-index:1.0.0-snapshot` on the Kogito CR
```
Comment (radtriste) [inline RELEASE_NOTES.md:4]:
```suggestion
     - To switch to Mongodb persistence provider just provide the `quay.io/kiegroup/kogito-data-index-mongodb:1.0.0-snapshot` on the Kogito CR
```
?

Comment (radtriste) [reply RELEASE_NOTES.md:4]: and provide MongoDB env var ?

**Review by radtriste — APPROVED** [review]

File: RELEASE_NOTES.md (hunk 2 above)
Comment (ricardozanini) [reply RELEASE_NOTES.md:4]: I believe we need to write specific documentation of how to connect the Data Index to MongoDB using java properties via `spec.config` or `spec.propertiesConfigMap` parameter.

**Review by ricardozanini — APPROVED** [review]
LGTM! Just waiting for @radtriste and @sutaakar comments to be addressed. :)

File: RELEASE_NOTES.md (hunk 2 above)
Comment (spolti) [reply RELEASE_NOTES.md:4]: I'll add it the docs, just trying to make it work on ocp yet, having hard times to find a operator easy to use, the only one for now is the percona mongodb community operator.

**Comment by ricardozanini** [comment]
@radtriste do we have time to make this one for 0.17?

**Comment by radtriste** [comment]
I don't think so. Release is starting on Monday morning.
We would need also to update BDD tests on operator side. @sutaakar do you confirm ?
I'd rather be safe on 0.17.0 as 1.0 will come soon after ;)

**Comment by sutaakar** [comment]
From BDD perspective we would have to figure out some usable MongoDB operator first, to have something to test against.
Or we can release it untested, or tested just manually locally.

**Comment by spolti** [comment]
@sutaakar @radtriste its up to you guys to decide if we can make ir or not for 0.17.
I have a environment with a deployed Kogito app + data-index with mongodb.

**Comment by radtriste** [comment]
@spolti 0.17 release is on Monday. I would rather be on the safe and waiting for this to be fully tested ... Ideally wait for having automated BDD tests for that. That could be done during the next week.
1.0 release is coming short after 0.17 (cut-off in 2 weeks) so I guess it can wait until then.

**Comment by radtriste** [comment]
and also, having https://issues.redhat.com/browse/KOGITO-3092 with that one would be great for the UX experience ;)
created https://issues.redhat.com/browse/KOGITO-3689 for the automation

**Comment by radtriste** [comment]
@spolti @ricardozanini should we merge this ?

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**